File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int answer = 0;
        
        int plus = common[1]-common[0];
        int multi = 0;
        if(common[0] !=0){
            multi = common[1]/common[0];
        }
        
        boolean check = false;
        
        for(int i = 1 ; i < common.length ; i++){
            if(common[i] - common[i-1] != plus){
                check = true;
                break;
            }
        }
        
        if(check)answer = common[common.length-1] * multi;
        else answer = common[common.length-1] +plus;
        
        
        
        return answer;
    }
}

File: C#/프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.cs
using System;

public class Solution {
    public int solution(int[] common) {
            int answer = 0;

            if (common[1] - common[0] == common[2] - common[1]) answer = common[0] + (common[1] - common[0]) * common.Length;
            else
            {
                int ratio = common[1] / common[0];
                answer = common[common.Length - 1] * ratio;
            }


            return answer;
    }
}